var/spack/repos/builtin/packages/intel-parallelstudio/package.py

```diff
@@ -0,0 +1,112 @@
+            base_components = "ALL"
+        else:
+            with open("pset/mediaconfig.xml", "r") as f:
+                lines = f.readlines()
+                all_components = []
+                for line in lines:
+                    if line.find('<Abbr>') != -1:
+                        component = line[line.find('<Abbr>') + 6:line.find('</Abbr>')]
+                        all_components.append(component)
+                base_components = filter_pick(all_components, re.compile('(comp|openmp|intel-tbb|icc|ifort|psxe|icsxe-pset)').search)
+                mpi_components = filter_pick(all_components, re.compile('(icsxe|imb|mpi|itac|intel-tc|clck)').search)
+                mkl_components = filter_pick(all_components, re.compile('(mkl)').search)
+                daal_components = filter_pick(all_components, re.compile('(daal)').search)
+                ipp_components = filter_pick(all_components, re.compile('(ipp)').search)
+                tool_components = filter_pick(all_components, re.compile('(gdb|vtune|inspector|advisor)').search)
+
+        components = base_components
+        if not spec.satisfies('+all'):
+            if spec.satisfies('+mpi') and 'cluster' in str(spec.version):
+                components += mpi_components
+            if spec.satisfies('+mkl'):
+                components += mkl_components
+            if spec.satisfies('+daal'):
+                components += daal_components
+            if spec.satisfies('+ipp'):
+                components += ipp_components
+            if spec.satisfies('+tools') and (spec.satisfies('@cluster') or spec.satisfies('@professional')):
+                components += tool_components
+
+        self.intel_components = ';'.join(components)
+        IntelInstaller.install(self, spec, prefix)
+
+        absbindir = os.path.split(os.path.realpath(os.path.join(self.prefix.bin, "icc")))[0]
+        abslibdir = os.path.split(os.path.realpath(os.path.join(self.prefix.lib, "intel64", "libimf.a")))[0]
+
+        # symlink or copy?
+        os.symlink(self.global_license_file, os.path.join(absbindir, "license.lic"))
+        if spec.satisfies('+tools') and (spec.satisfies('@cluster') or spec.satisfies('@professional')):
+            os.mkdir(os.path.join(self.prefix, "inspector_xe/licenses"))
+            os.symlink(self.global_license_file, os.path.join(self.prefix, "inspector_xe/licenses", "license.lic"))
+            os.mkdir(os.path.join(self.prefix, "advisor_xe/licenses"))
+            os.symlink(self.global_license_file, os.path.join(self.prefix, "advisor_xe/licenses", "license.lic"))
+            os.mkdir(os.path.join(self.prefix, "vtune_amplifier_xe/licenses"))
+            os.symlink(self.global_license_file, os.path.join(self.prefix, "vtune_amplifier_xe/licenses", "license.lic"))
+
+        if (spec.satisfies('+all') or spec.satisfies('+mpi')) and spec.satisfies('@cluster'):
+            os.symlink(self.global_license_file, os.path.join(self.prefix, "itac_latest", "license.lic"))
```

Comment:
@adamjstewart This gets a bit messy if those tools aren't installed because we create subdirectories for them, which may confuse users. Unfortunately, we don't know until install time which tools are actually installed, and since the install is done in a forked environment, we can't modify the license_files in a way that the post_install hook will see the modifications. Are you strongly opposed to the manual creation of symlinks? Got any other ideas on how we can do this?